mathesar_ui/src/systems/data-explorer/displayOptions.ts

```diff
@@ -0,0 +1,336 @@
+  index: number;
+  type: {
+    name: string;
+    item_type?: string;
+  };
+}
+
+function anchorTypeEq(
+  a: ColumnAnchor['type'],
+  b: ColumnAnchor['type'],
+): boolean {
+  if (a.name !== b.name) return false;
+  if (a.item_type !== b.item_type) return false;
+  return true;
+}
+
+export interface ColumnDisplayOptionsEntry {
+  column: ColumnAnchor;
+  displayOptions: ColumnMetadata;
+}
+
+/**
+ * Keys are the stringified indexes (zero-based) of the column in the
+ * exploration result set.
+ */
+type ColumnDisplayOptions = Record<string, ColumnDisplayOptionsEntry>;
+
+/**
+ * Instances of this type get written to Explorations.display_options in the
+ * internal database, but the back end never uses it.
+ *
+ * ⚠️ Beware that breaking changes to this type (and contained types) might
+ * require a migration step in order to preserve functionality for
+ * previously-saved explorations.
+ *
+ * That's why we have a child property that might seem to present unnecessary
+ * indirection — the nesting exists to allow us to add more properties here
+ * without making breaking changes.
+ */
+export interface ExplorationDisplayOptions {
+  columnDisplayOptions?: ColumnDisplayOptions | null;
+}
+
+export function validateDisplayOptions(
+  input: AddableExploration['display_options'],
+): ExplorationDisplayOptions {
+  if (!input) {
+    return {
+      columnDisplayOptions: {},
+    };
+  }
+
+  return input as ExplorationDisplayOptions;
+
+  // You'll notice this function is not actually doing any real "validation"
+  // right now. So why does it exist?
+  //
+  // The exploration display options represent a rare case of a non-trivial data
+  // structure that the front end "owns" but that gets stored in the service
+  // layer. The back end has no mechanism to validate or use exploration display
+  // options. These options only ever generated by the front end and only ever
+  // used by the front end.
+  //
+  // So if we decide to change this data structure in the future, then we might
+  // need some sort of migration system so that an exploration which the user
+  // generated and saved using old Mathesar code can still be used with new
+  // Mathesar code.
+  //
+  // For these reasons, the display options types are loosely coupled with the
+  // API types. We want to encapsulate all of the display options complexity
+  // within this module, while still keeping the API types low-level, and
+  // dependent primarily on _back-end_ code.
+  //
+  // There may come a day when we need more sophisticated code to handle
+  // breaking changes to display options data structures.
+  //
+  // So for now, this function exists to serve as a clear and singular barrier
+  // between the un-typed display options at the API layer and the
+  // strongly-typed display options in the front end.
+}
+
+export function makeColumnAnchor(
+  resultColumn: QueryResultColumn,
+  index: number,
+): ColumnAnchor {
+  return {
+    name: resultColumn.alias,
```

Comment:
The term `name` here is confusing for me since we use the term alias everywhere else, and `name` is ambiguous and could refer to the alias, display name, and the column's actual name.